soc/arm/st_stm32/stm32f0/soc.c

```diff
@@ -36,7 +39,9 @@
  */
 void relocate_vector_table(void)
 {
-#ifndef CONFIG_IS_BOOTLOADER
+#if defined(CONFIG_SW_VECTOR_RELAY) || defined(CONFIG_SW_VECTOR_RELAY_CLIENT)
+	_vector_table_pointer = _vector_start;
+#elif !defined(CONFIG_IS_BOOTLOADER)
```

Comment:
I assume this fails in case the application is started by a bootloader and and the application sets CONFIG_SW_VECTOR_.... In this scenario the vector table in use is the vector table in FLASH from the bootloader. The vector table of the application (also in FLASH) is just not used.

Reply:
SW_VECTOR_RELAY is an option to be used by bootloaders; we specify this is the Kconfig definition.
SW_VECTOR_RELAY_CLIENT shall only be used on images that are booted by a bootloader which has been built using SW_VECTOR_RELAY; we also specify this.

So, right, in your scenario the application will fail but this is really a misuse of the feature.

Reply:
Thanks for the explanation. Didn´t know that.  

> So, right, in your scenario the application will fail but this is really a misuse of the feature.

Going after your explanation the application will only fail if the bootloader did _NOT_ set CONFIG_SW_VECTOR_RELAY=y.

I think this needs some in code commenting to make the whole thing understandable.

Reply:
I think that the fact that having SW_VECTOR_RELAY and SW_VECTOR_RELAY_**CLIENT**, makes it intuitive that these should be used coupled with each-other. But sure, I am all up for more enhanced code commenting, either here, or in the help text of the Kconfig definitions of these options. Ur welcome to send an enhancement PR.